### ISSUE TITLE
express app bundles incorrectly

### ISSUE THREAD

**Issue by peterwangfeng** [oven-sh/bun#2911]
### What version of Bun is running?

0.6.1

### What platform is your computer?

ubuntu 22.04 x86_64

### What steps can reproduce the bug?

I have a test.js, and code as below:
```ts
import express from "express";

const app = express();
const port = 8081;

app.get("/", (req, res) => {
	res.send("Hello World!");
});

app.listen(port);

```
l want use bun build --compile test.js get test  standalone executables and then execute ./test in terminal
open browser http://localhost:8081/

### What is the expected behavior?

![image](https://github.com/oven-sh/bun/assets/23310431/89b0045a-d9d6-47ee-8124-87ca23a83304)

### What do you see instead?

![image](https://github.com/oven-sh/bun/assets/23310431/42775a81-f552-4058-adbc-f95a86a8643d)

**Comment by paperclover** [comment]
reproduced on linux. can be reproduced with just `bun build`. side note is we should implement err.stack to match how node works, but this is the error we get

![image](https://github.com/oven-sh/bun/assets/24465214/1e2a6de1-5ed0-47f4-8b9f-088939088fba)

**Comment by paperclover** [comment]
Minimal Reproduction
```ts
const fn = require('fresh');
console.log(fn());
```

in `node_modules/fresh/index.js`:
```ts
module.exports = function() {
  return 'it worked';
}
```

bundle:
![image](https://github.com/oven-sh/bun/assets/24465214/96002212-a874-4647-b92c-93f9e7327058)

the issue is that we are adding `__toESM` but only in the node_modules case. not sure why.

**Comment by paperclover** [comment]
Fixed in #3183
